pwndbg/gdblib/stack.py

```diff
@@ -136,3 +136,27 @@ def is_executable() -> bool:
             nx = True
 
     return not nx
+
+
+def find_addr_on_stack(addr) -> int:
```

Comment:
I am a bit worried that this may be (way) too slow.

* We are reading the whole stack memory: what if stack is super big?
* What if SP does not point to mapped memory? This will crash
* This can be optimized out in two ways:
1) We could leverage the existing search functionality we have which will do a find using GDB's API which is written in C and so should be faster
2) We could read the whole stack memory and use simple `.find(...)` to find the offset of the bytes of the `addr` value

The 2) will rather be slower than 1) but if we read the stack memory just once, then we can find the fetched bytes multiple times to look for different return addresses --- which would probably be faster overall?

Reply:
Given this I am not sure if we really want to look for those values on the stack... we are making context slower this way. We can however make this configurable.

@gsingh93 thoughts?